File: components/catalogs/templates/refactor.py
import os
import re

DIR = r"c:\Users\Mehme\.gemini\antigravity\scratch\katalog-app\components\catalogs\templates"

for filename in os.listdir(DIR):
    if not filename.endswith('.tsx') or filename in ['index.tsx', 'index.ts', 'catalog-template-renderer.tsx', 'utils.tsx', 'types.ts', 'types.tsx']:
        continue

    filepath = os.path.join(DIR, filename)
    with open(filepath, 'r', encoding='utf-8') as f:
        content = f.read()

    original_content = content
    
    # 1. Update imports
    if 'getStandardLogoHeight' not in content:
        content = re.sub(
            r'import\s+\{([^}]+)\}\s+from\s+["\']\./utils["\']',
            r'import { \1, getStandardLogoHeight, getHeaderLayout } from "./utils"',
            content
        )
    
    # 2. Extract out completely the `getLogoHeight = () => { switch (logoSize) ... }`
    content = re.sub(
        r'\s*const\s+(getLogoHeight|getLogoDimensions)\s*=\s*\(\)\s*=>\s*\{\s*switch\s*\(logoSize\)\s*\{\s*(case\s+\'\w+\'\s*:\s*return[^}]+)*default\s*:\s*return[^}]+\}\s*\}',
        '',
        content,
        flags=re.DOTALL
    )

    # 3. Replace the layout logic
    regex_layout = re.compile(
        r'\s*const\s+isHeaderLogo\s*=\s*logoPosition\?\.startsWith\(\'header\'\).*?'
        r'(?:const\s+logoAlignment[^\n]*\n)?'
        r'(?:.*?const\s+isCollisionLeft\s*=\s*[^\n]*\n)?'
        r'(?:.*?const\s+isCollisionCenter\s*=\s*[^\n]*\n)?'
        r'(?:.*?const\s+isCollisionRight\s*=\s*[^\n]*\n)?'
        r'(?:.*?const\s+isAnyCollision\s*=\s*[^\n]*\n)?',
        re.DOTALL
    )

    replacement = """
    const { isHeaderLogo, logoAlignment, titlePosition: finalTitlePosition, isCollisionLeft, isCollisionCenter, isCollisionRight, isAnyCollision } = getHeaderLayout(logoPosition, titlePosition)
    const logoHeight = getStandardLogoHeight(logoSize)
"""
    
    match = regex_layout.search(content)
    if match and "getHeaderLayout" not in content[match.start():match.end()]:
        content = content[:match.start()] + replacement + content[match.end():]

    # 4. Replace usages of getLogoHeight() with logoHeight
    content = content.replace("getLogoHeight()", "logoHeight")
    
    # Specific fix for files using destructured getLogoDimensions()
    content = content.replace("const { height, className } = getLogoDimensions()", "const height = logoHeight\n        const className = 'object-contain max-h-12 w-auto'")
    
    # Clean up double imports
    content = content.replace(", getStandardLogoHeight, getHeaderLayout, getStandardLogoHeight, getHeaderLayout", ", getStandardLogoHeight, getHeaderLayout")
    
    if content != original_content:
        with open(filepath, 'w', encoding='utf-8') as f:
            f.write(content)
        print(f"Updated {filename}")
    else:
        print(f"No changes made to {filename}")

EOF

File: components/catalogs/templates/refactor.mjs
import fs from 'fs';
import path from 'path';
import { fileURLToPath } from 'url';

const __filename = fileURLToPath(import.meta.url);
const __dirname = path.dirname(__filename);

const templatesDir = __dirname;
const files = fs.readdirSync(templatesDir).filter(f => f.endsWith('.tsx') && !f.includes('types'));

for (const file of files) {
    const filePath = path.join(templatesDir, file);
    let content = fs.readFileSync(filePath, 'utf-8');
    let changed = false;

    if (!content.includes('getHeaderLayout')) {
        content = content.replace(
            /import { buildBackgroundStyle, sanitizeHref, formatProductPrice } from "\.\/utils"/,
            `import { buildBackgroundStyle, sanitizeHref, formatProductPrice, getStandardLogoHeight, getHeaderLayout } from "./utils"`
        );
        changed = true;
    }

    // Attempt to match the old getLogoHeight to isCollisionRight block
    const oldBlockRegex = /const getLogoHeight = \(\) => {[\s\S]*?const isCollisionRight = [^\n]*\n/;

    if (oldBlockRegex.test(content)) {
        content = content.replace(oldBlockRegex, `    const {
        isHeaderLogo,
        logoAlignment,
        titlePosition: finalTitlePosition,
        isCollisionLeft,
        isCollisionCenter,
        isCollisionRight
    } = getHeaderLayout(logoPosition, titlePosition)

    const logoHeight = getStandardLogoHeight(logoSize)\n`);
        changed = true;
    }

    if (content.includes('getLogoHeight()')) {
        content = content.replace(/getLogoHeight\(\)/g, 'logoHeight');
        changed = true;
    }

    if (changed) {
        fs.writeFileSync(filePath, content, 'utf-8');
        console.log(`Updated ${file}`);
    }
}
